### PR TITLE
feat((http)): Add proxy support when using the tool

### LINKED ISSUE #78: Allow for a proxy workflow
The SDKs support proxy usage but there's no way to pass those from the tool.

### LINKED ISSUE #93: Not able to run the sync process
I followed all the steps mentioned in https://www.contentful.com/developers/docs/tutorials/general/using-contentful-space-sync/

but This is what I am getting. Not helpful error message at all!

Command : `contentful-space-sync --config my-config-file.json`

Output: 
`info Contentful Space Sync:`
`Let's sync some content across spaces!`
`info No previous sync token found.`
`Synchronizing fresh content from xxxxxxxx to yyyyyyyyyyy`
`info Getting content from source space via the sync api`
`ERR! Failed with`
 `undefined`

Here is how my config file looks like:

`{
  "sourceSpace": "xxxxxxxxx",
  "destinationSpace": "yyyyyyyyyyyy",
  "sourceDeliveryToken": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
  "managementToken": "yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy"
}`

Both spaces are within the same org and I have access to both. 

No log files are generated either.
Please help.

### PR DESCRIPTION
**Pull request by Khaledgarbaya**
The SDKs support proxy usage but there was no way to pass those from the tool before.

closes #78, closes #93

### CODE REVIEW

File: lib/run-space-sync.js [hunk 1]
```diff
@@ -13,7 +13,10 @@ import dumpErrorBuffer from './dump-error-buffer'
 import getTransformedDestinationResponse from './get-transformed-destination-response'
 
 export default function runSpaceSync (usageParams) {
-  const {opts, syncTokenFile, errorLogFile} = usageParams
+	const {opts, syncTokenFile, errorLogFile} = usageParams
+	if(opts.proxyHost && opts.proxyPort){
+		opts.proxy = {host: opts.proxyHost, port: opts.proxyPort}
+	}
   const clients = createClients(opts)
```
Comment (stefanjudis) [inline lib/run-space-sync.js]: here is something weird, or?
The new lines are slightly off...

File: lib/usageParams.js [hunk 2]
```diff
@@ -76,8 +76,26 @@ var opts = yargs
...
+		if (!argv.proxyHost && !argv.proxyPort){
+			return true  
+		}
+		if (argv.proxyPort && argv.proxyHost){
+			return true
+		}
+		log.error('--proxy-host and --proxy-port must be both defined')
+		process.exit(1)
+	})
   .check(function (argv) {
```
Comment (stefanjudis) [inline lib/usageParams.js]: Same, weird off spacing.

File: lib/run-space-sync.js (hunk 1 above)
Comment (Khaledgarbaya) [inline lib/run-space-sync.js]: yeah that's why the build failed, some wrong when with vim :(

File: lib/usageParams.js (hunk 2 above)
Comment (Khaledgarbaya) [inline lib/usageParams.js]: Same 😭

**Comment by stefanjudis** [comment]
@Khaledgarbaya apart from the formatting it looks good to me. 👍

**Pull request merged**